### PR TITLE
feat: 1653 first feature flag poc

### PR DESCRIPTION
**Pull request by frankvonhoven**
## **Description**

* This is the actual first implementation of the LaunchDarkly API for feature flags
* This utilizes the `mobileMinimumVersions.appMinimumBuild` feature flag in LaunchDarkly 

## **Related issues**

Fixes: [#1653](https://github.com/MetaMask/mobile-planning/issues/1653)

This PR builds upon work in previous LD setup work: [#10294](https://github.com/MetaMask/metamask-mobile/pull/10294)

## **Manual testing steps**

1. Start the app
2. in `app/components/hooks/MinimumVersions/useMinimumVersions.tsx`
  * Change line: 22 `featureFlags?.mobileMinimumVersions?.appMinimumBuild > currentBuildNumber;`
        to: `featureFlags?.mobileMinimumVersions?.appMinimumBuild > 0;`
3. Restart app & enter password
4. Modal should popup prompting user to update app version

### **After**

https://github.com/user-attachments/assets/812fb49d-e79b-4b3f-8f5e-00d3b63a431f

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by NicolasMassart — COMMENTED** [review]
We need tests!

File: app/components/hooks/MinimumVersions/useMinimumVersions.test.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }));
+
+    (getBuildNumber as jest.Mock).mockReturnValue('101');
+
+    renderHook(() => useMinimumVersions());
+
+    expect(mockNavigation.navigate).not.toHaveBeenCalled();
+  });
+
+  it('should not navigate if currentBuildNumber is greater than appMinimumBuild', () => {
```
Comment (NicolasMassart) [inline app/components/hooks/MinimumVersions/useMinimumVersions.test.ts:54]:
```suggestion
  it('requires update only if currentBuildNumber is lower than appMinimumBuild', () => {
```

**Review by NicolasMassart — COMMENTED** [review]
Simplify test name so they are easy to understand the use case. People hardly think in reverse.
Direct positive sentences are easier even if implementation requires an expect.not.to...

**Review by tommasini — DISMISSED** [review]
LGTM! Aggree with Nico comments! let me know when resolved and I will approve it again

**Review by tommasini — APPROVED** [review]

**Pull request merged**